IdentityCore/src/cache/MSIDTokenCache.h

```diff
@@ -25,33 +25,86 @@
 
 @class MSIDToken;
 @class MSIDTokenCacheKey;
+@class MSIDUser;
+
+@protocol MSIDTokenCacheDataSource;
 
 @interface MSIDTokenCache : NSObject
 
-- (BOOL)saveToken:(MSIDToken *)token
-           forKey:(MSIDTokenCacheKey *)key
-            error:(NSError **)error;
+- (id)initWithDataSource:(id<MSIDTokenCacheDataSource>)dataSource;
+- (id<MSIDTokenCacheDataSource>)dataSource;
 
-- (MSIDToken *)getAdalAT:(NSString *)userId
-                resource:(NSString *)resource
-                clientId:(NSString *)clientId
-                   error:(NSError **)error;
+/*!
+ Returns a AT/RT Token Cache Item for the given parameters. The RT in this item will only be good
+ for the given resource. If no RT is returned in the item then a MRRT or FRT should be used (if
+ available).
+ */
+- (MSIDToken *)getAdalATRTforUser:(MSIDUser *)user
```

Comment:
what's the difference between this one and getAdfsUserTokenForAuthority? For ADFS this one should still work, but return both AT and RT?

Reply:
1. we 'could' collapse it, but with adfs case being special, i left it out separated. we could discuss about this.
2. we would need a third type of token type for adfs token as it could contain both at and rt